apps/pulseprice/pulseprice.star

```diff
@@ -0,0 +1,193 @@
+iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAABEUlEQVQY012QP0vD
+UBRHz315AUswwTgIUnBQnDI42LkiQgVdi1+g0NFNipMO0gpuTiq4tF9AqIs42EEQ
+rFvBQQSRrqJRh/xB6/AaAl74LZfDGY6Q3ehoE6xl0JgpgD5S7wKIgfaKYA/A9nJQ
+AyoECZD6UBld2qzhO9esUUGzjo4iapyw6MBP0xhH22Wwb0Dzzu7nA0/uNC4+LnO0
+x2ZZURC3IAZizug6s/hvSyxwwEWY/SFtKUiACIjY4c5ymJjq8/hxyouXgwkK0kZu
+LH/N4KtJCt4hATkYNxTS7kHSWcX53aJSuOI2eeb1u8R8bKCog+z3xnmqRZABqH95
+7BB0gBwPJQ9e3QBKJnQGWvfI+SXAH33pUizRrvIoAAAAAElFTkSuQmCC
+""")
+
+NO_DATA = "---------- "
+
+DEXSCREENER_PLS_URL = "https://api.dexscreener.com/latest/dex/tokens/0xA1077a294dDE1B09bB078844df40758a5D0f9a27"
+DEXSCREENER_INC_URL = "https://api.dexscreener.com/latest/dex/tokens/0x2fa878Ab3F87CC1C9737Fc071108F904c0B0C95d"
+
+def main():
+    # Get PLS, PLSX price
+    pls_price_data = get_json_from_cache_or_http(DEXSCREENER_PLS_URL, ttl_seconds = 600)
+
+    pls_price = NO_DATA
+    plsx_price = NO_DATA
+
+    if pls_price_data != None:
+        pls_price = "$" + pls_price_data["pairs"][0]["priceUsd"]
+        plsx_price = "$" + pls_price_data["pairs"][2]["priceUsd"]
+
+    # IN price
+    inc_price_data = get_json_from_cache_or_http(DEXSCREENER_INC_URL, ttl_seconds = 600)
+
+    inc_price = NO_DATA
+
+    if inc_price_data != None:
+        inc_price = "$" + inc_price_data["pairs"][0]["priceUsd"]
+
+    # Setup display rows
+    displayRows = []
+
+    displayRows.append(
+        render.Row(
+            expanded = True,
+            main_align = "space_evenly",
+            cross_align = "center",
+            children = [
+                render.Image(src = PLS_ICON_SM),
+                render.Text(pls_price),
+            ],
+        ),
+    )
+
+    displayRows.append(
+        render.Row(
+            expanded = True,
+            main_align = "space_evenly",
+            cross_align = "center",
+            children = [
+                render.Image(src = PLSX_ICON_SM),
+                render.Text(plsx_price),
+            ],
+        ),
+    )
+
+    displayRows.append(
+        render.Row(
+            expanded = True,
+            main_align = "space_evenly",
+            cross_align = "center",
+            children = [
+                render.Image(src = INC_ICON_SM),
+                render.Text(inc_price),
+            ],
+        ),
+    )
+
+    return render.Root(
+        child = render.Stack(
+            children = [
+                render.Column(
+                    main_align = "space_evenly",  # this controls position of children, start = top
+                    expanded = True,
+                    cross_align = "center",
+                    children = displayRows,
+                ),
+            ],
+        ),
+    )
+
+def get_json_from_cache_or_http(url, ttl_seconds):
+    cached_response = cache.get(url)
+
+    if cached_response != None:
+        print("Cache hit: {}".format(url))
+        data = json.decode(cached_response)
+    else:
+        print("HTTP JSON Request: {}".format(url))
+        http_response = http.get(url)
```

Comment:
Check out the http modules new(ish) cache support. Should simplify this block of code a bit, and also future proof the app against potential future change to the cache module. Read more [over here](https://discuss.tidbyt.com/t/http-caching-client/5129)